sharding/contracts/sharding_manager.sol

```diff
@@ -221,4 +238,22 @@ contract SMC {
         --emptySlotsStackTop;
         return emptySlotsStack[emptySlotsStackTop];
     }
-}
\ No newline at end of file
+
+    /// Check if a bit is set, this function is used to check
+    /// if a notary has casted the vote, return true if voted, false if not
+    function hasVoted(uint _shardId, uint _index) internal returns (bool) {
+        return currentVote[_shardId] & bytes32(ONE << uint(ONES - _index)) == bytes32(ONE << uint(ONES - _index));
+    }
+
+    /// Set a bit to one, notary uses this function to cast it's vote,
+    /// After the notary casts it's vote, we increase the vote count by 1 (last byte)
+    /// Returns the total numbers of votes
+    function castVote(uint _shardId, uint _index) internal returns (uint8) {
```

Comment:
Can you add more to the description of voting?

How is this supposed to work? Are there only 31 notaries and they place their vote on their index?

Reply:
I'll add more color to the comments. There are 31 * 8 notaries. Each place it's vote on the unique index.

Reply:
Ah, yes that would be helpful context